tests/compiler/test_typing.py

```diff
@@ -103,31 +103,162 @@ def test_int_type_invalid(snippetcompiler):
     )
 
 
+def test_number_type(snippetcompiler):
+    snippetcompiler.setup_for_snippet(
+        """
+entity Test:
+    number i = 0
+end
+
+implement Test using std::none
+
+Test(i = 42)
+Test(i = -42)
+Test()
+Test(i = 42.0)
+Test(i = false)
+        """,
+    )
+    compiler.do_compile()
+
+
 def test_cast_to_number(snippetcompiler):
     snippetcompiler.setup_for_snippet(
         """
-x = number("42")
-x = 42
-y = number("42.0")
-y = 42.0
+a = number(21)
+a = 21.0
+b = number(25.0)
+b = 25.0
+x = number("31")
+x = 31.0
+y = number("22.0")
+y = 22.0
 z = number(true)
-z = 1
+z = 1.0
 u = number(false)
-u = 0
+u = 0.0
         """,
     )
     (_, scopes) = compiler.do_compile()
     root: Namespace = scopes.get_child("__config__")
+    a = root.lookup("a").get_value()
+    b = root.lookup("b").get_value()
     x = root.lookup("x").get_value()
     y = root.lookup("y").get_value()
     z = root.lookup("z").get_value()
     u = root.lookup("u").get_value()
+    assert Number().validate(a)
+    assert Number().validate(b)
     assert Number().validate(x)
     assert Number().validate(y)
     assert Number().validate(z)
     assert Number().validate(u)
 
 
+def test_float_alias_number(snippetcompiler):
```

Comment:
Could you add some lines verifying that the two types are compatible? i.e. `test.my_float = test.my_int` and vice versa.

Also, `int <- float` and `float <- int` assignments would be useful. The former should raise an error while the latter should work and result in a float.